### ISSUE TITLE
Don't rely on external commands

### ISSUE THREAD

**Issue by piranna** [lovell/detect-libc#1]
Allow to detect system libc on systems that don't have `getconf` or `ldd` commands, like NodeOS. This would be done by reading the symlink at `/lib/libc.so`.

**Comment by lovell** [comment]
Hello, are you able to provide a (partial) code sample or worked example to help me understand how the symlink will help determine libc family?

**Comment by piranna** [comment]
```sh
ls -l /lib/ld-linux.so.2 
lrwxrwxrwx 1 root root 25 jun 16 20:42 /lib/ld-linux.so.2 -> i386-linux-gnu/ld-2.24.so
```

That's for Ubuntu GNU system, for musl systems the symlink path will by `*-musl`. In NodeOS code there are some similar checks for this and for the ARM different floating point architectures.

**Comment by lovell** [comment]
Thank you. I fired up a few different Linux VMs and the presence of a `/lib/*-linux-gnu` directory seems to indicate glibc and the presence of a `/lib/libc.musl-*` or `/lib/ld-musl-*` file indicates musl. This should be enough for some filesystem-based checks to determine the libc family (but perhaps not the version) should `getconf` etc. fail.

**Comment by lovell** [comment]
Here's what I see for NodeOS, which suggests the check for `/lib/ld-musl-*` will work.
```sh
$ docker run --rm -t -i nodeos/nodeos
Starting REPL session
NodeOS-boot-multiuser> fs.readdirSync('/lib')
[ 'ld-musl-x86_64.so.1',
  'libc.so',
  'libfuse.so',
  'libfuse.so.2',
  'libfuse.so.2.9.7',
  'libgcc_s.so.1',
  'libstdc++.so.6',
  'libstdc++.so.6.0.21',
  'node_modules' ]
```

**Comment by piranna** [comment]
Then you have the answer :-)

**Comment by lovell** [comment]
Thanks, support for filesystem-based family detection is now available in v0.2.0.
